indexer/services/roundtable/src/tasks/aggregate-trading-rewards.ts

```diff
@@ -213,23 +229,305 @@ export class AggregateTradingReward {
+  }
+
+  private async addEndedAtAndEndedAtHeightUpdates(
+    aggregationUpdateAndCreateObjects: AggregationUpdateAndCreateObjects,
+    interval: Interval,
+  ): Promise<AggregationUpdateAndCreateObjects> {
+    const startedAt: string = this.getStartedAt(interval);
+    const startedAtHeight: string = await this.getNextBlock(startedAt);
+    const endedAt: IsoString = this.getEndedAt(interval);
+    // endedAtHeight is the first block created before endedAt
+    const endedAtHeight: string = Big(await this.getNextBlock(endedAt)).minus(1).toFixed();
+
+    const allIncompleteAggregation: TradingRewardAggregationFromDatabase[] = await
+    TradingRewardAggregationTable.findAll({
+      period: this.period,
+      startedAtHeight,
+    }, []);
+    const allIncompleteAggregationAddresses: string[] = _.map(
+      allIncompleteAggregation,
+      TradingRewardAggregationColumns.address,
+    );
+
+    const otherAggregationAddresses: string[] = _.difference(
+      allIncompleteAggregationAddresses,
+      _.map(
+        aggregationUpdateAndCreateObjects.updateObjects,
+        TradingRewardAggregationColumns.address,
+      ),
+    );
+    const otherAggregationUpdateObjects: TradingRewardAggregationUpdateObject[] = _.map(
+      otherAggregationAddresses,
+      (address: string) => {
+        return {
+          id: TradingRewardAggregationTable.uuid(address, this.period, startedAtHeight),
+          endedAt,
+          endedAtHeight,
+        };
+      },
+    );
+    return {
+      createObjects: _.map(
+        aggregationUpdateAndCreateObjects.createObjects,
+        (createObject: TradingRewardAggregationCreateObject) => {
+          return {
+            ...createObject,
+            endedAt,
+            endedAtHeight,
+          };
+        },
+      ),
+      updateObjects: _.map(
+        aggregationUpdateAndCreateObjects.updateObjects,
+        (updateObject: TradingRewardAggregationUpdateObject) => {
+          return {
+            ...updateObject,
+            endedAt,
+            endedAtHeight,
+          } as TradingRewardAggregationUpdateObject;
+        },
+      ).concat(otherAggregationUpdateObjects),
+    };
+  }
+
+  private getEndedAt(interval: Interval): IsoString {
+    const startedAt: IsoString = this.getStartedAt(interval);
+
+    return DateTime
+      .fromISO(startedAt, UTC_OPTIONS)
+      .endOf(this.getDateTimeUnit())
+      .plus({ milliseconds: 1 })
+      .toISO();
+  }
+
+  private getDateTimeUnit(): DateTimeUnit {
+    switch (this.period) {
+      case TradingRewardAggregationPeriod.DAILY:
+        return DateTimeUnit.day;
+      case TradingRewardAggregationPeriod.WEEKLY:
+        return DateTimeUnit.week;
+      case TradingRewardAggregationPeriod.MONTHLY:
+        return DateTimeUnit.week;
+      default:
+        throw new Error(`Invalid period ${this.period}`);
+    }
+  }
+
+  private async setAggregationUpdateAndCreateObjects(
```

Comment:
consider renaming to `createAndUpdateAggregations`

Reply:
I like this name more, renamed